h_program-lang/AST_generic.ml

```diff
@@ -947,6 +947,7 @@ and type_ =
   | TyNameApply of name * type_arguments
 
   | TyVar of ident (* type variable in polymorphic types (not a typedef) *)
+  | TyAny of ident (* anonymous type, '_' in OCaml *)
```

Comment:
could be TyAny of tok then, if you know it's always _, there is no need for the ident value.

Reply:
I thought we might find the same feature in another language, where the symbol is not `_`.